Root/OverlapTools.cxx

```diff
@@ -373,6 +385,287 @@ void OverlapTools::j_t_overlap(TauVector& taus, JetVector& jets, double dR)
         } // for(iTau)
     } // for(iJet)
 }
+
+
+//----------------------------------------------------------
+// begin OverlapTools_4Lep
+//----------------------------------------------------------
+void OverlapTools_4Lep::performOverlap(ElectronVector& electrons, MuonVector& muons,
+                                    TauVector& taus, JetVector& jets)
+{
+						///without the Photon OR
+                                                /// updated for SUSYTools/harmonization consistency Feb/16
+  /*1*/	t_e_overlap(taus,electrons,0.2); // straightforward dR comparison
+  /*2*/	t_m_overlap(taus, muons,0.2);    // dR comparison for taus pT<50. For tau pT>50, only compare to combined muons
+  /*3*/ e_m_overlap(electrons,muons);    // discard electrons sharing ID track with muons (Calo muons are be removed first if they share a track with an electron)
+  ///*4,5*/ photon_lep_overlap(photons, electrons, muons, 0.4);	// discard pho overlapping with e/mu
+  /*6*/	j_e_overlap(electrons, jets, 0.2);  // discard non-b-tagged jets with dR<0.2
+        e_j_overlap(electrons,jets,0.4);    // discard ele with dR<0.4 to non-pileup jets
+
+  /*7*/	j_m_overlap(muons, jets,0.2);  // discard jets with few tracks or small jet/mu pT ratio. 
+        m_j_overlap(muons, jets,0.4);       // discard ele with dR<0.4 to non-pileup jets
+  
+  /*8*/	j_t_overlap(taus,jets,0.2);  // discard jets overlapping with signal taus
+
+	///*9*/ j_photon_overlap(photons, jets, 0.4);	// discard jet overlapping with photon
+
+	//DY is performed in  'm_nttools.removeSFOSPairs' step
+	
+}
+void OverlapTools_4Lep::performOverlap(ElectronVector& electrons, MuonVector& muons,
```

Comment:
@tinapotter I wonder if there is a way to combine the two "performOverlap" methods that will be able to handle both cases for the taus? If not that is perfectly fine of course -- these are your functions -- but having essentially duplicated methods increases the book-keeping and could come back to bite us in the future